src/nvim/api/win_config.c

```diff
@@ -212,6 +212,10 @@
 ///   - hide: If true the floating window will be hidden.
 ///   - vertical: Split vertically |:vertical|.
 ///   - split: Split direction: "left", "right", "above", "below".
+///   - cmdline: If true, indicate that this window is used for ext_cmdline.
+///     Nvim will anchor the 'wildoptions' popupmenu to this window.
+///   - cmdline_offset: Offset in units of "screen cell width" indicating how much longer
+///     (negative for shorter) the displayed prompt is from the prompt emitted by "cmdline_show".
```

Comment:
Could keep these private for use with the default ext_cmdline (`_cmdline/_cmdline_offset`) but after testing this seems useful enough to expose.

Reply:
If we expose these, they should be options (for reference, the existing 'previewwindow' option makes a similar kind of decision). Something like a new, global 'popupwindow' option (or 'popupopt', 'pmenuopt', ...).

Setting a global option from an unrelated "create"/"open" API is strange.

If it helps unblock this PR, for now let's just not expose these.

Reply:
Still awaiting @bfredl's review, who originally suggested adding this to `nvim_open_win` :) From matrix:

>> I was also wondering if some API to "set" the cmdline window would be better for some internal work Nvim can do. But perhaps documenting that the implementation should move the cursor there to this end is sufficient?

> I think this can be done, but being future-proof for more "roles" being added later. perhaps a field in nvim_win_set_config/win_open

Reply:
`nvim_win_set_config` would fit better, since it is essentially window-local properties, which is very similar to window-local options. Though a global option seems to make more sense than a window-local option, since a global option is easier to "query".

Reply:
`nvim_win_set_config` and `nvim_open_win` share the same `opts` dictionary, so adding it to one implies the other. I'm not sure how a global option would work, do you mean that option would hold the window handle?

Reply:
> do you mean that option would hold the window handle?

Yes. That seems natural, no?

Reply:
Yes if it's a global option it would have to be set/unset whenever the window is deleted (which the user could do, at which point the handler would have to create a new window). So unless we disallow deleting a window marked as such, attaching the flag to the window itself seems to make more sense to me.

Reply:
There is something to say for symmetry with `previewwindow` though, so should probably promote it to an actual window option if @bfredl agrees. This does bring into question the entire `WinConfig` struct.

Reply:
> This does bring into question the entire `WinConfig` struct.

Let's save that for https://github.com/neovim/neovim/issues/30453 , out of scope here.

Reply:
I'd say keep it a `_` field in winconfig for now.